src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/oracle/PMTransactionDataProvider.kt

```diff
@@ -0,0 +1,112 @@
+package it.pagopa.ecommerce.helpdesk.dataproviders.oracle
+
+import io.r2dbc.spi.ConnectionFactory
+import it.pagopa.ecommerce.helpdesk.dataproviders.TransactionDataProvider
+import it.pagopa.ecommerce.helpdesk.exceptions.InvalidSearchCriteriaException
+import it.pagopa.ecommerce.helpdesk.exceptions.NoResultFoundException
+import it.pagopa.ecommerce.helpdesk.utils.resultToTransactionInfoDto
+import it.pagopa.generated.ecommerce.helpdesk.model.*
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+import reactor.kotlin.core.publisher.toMono
+
+/**
+ * TransactionDataProvider implementation that search transactions into PM DB
+ *
+ * @see TransactionDataProvider
+ */
+@Component
+class PMTransactionDataProvider(@Autowired private val connectionFactory: ConnectionFactory) :
+    TransactionDataProvider {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    override fun totalRecordCount(searchCriteria: HelpDeskSearchTransactionRequestDto): Mono<Int> =
+        when (searchCriteria) {
+            is SearchTransactionRequestPaymentTokenDto -> Mono.just(0)
+            is SearchTransactionRequestRptIdDto -> Mono.just(0)
+            is SearchTransactionRequestTransactionIdDto -> Mono.just(0)
+            is SearchTransactionRequestEmailDto ->
+                getTotalResultCount(buildTransactionByUserEmailCountQuery(searchCriteria.userEmail))
+            is SearchTransactionRequestFiscalCodeDto ->
+                Mono.error(RuntimeException("Not implemented yet"))
+            else ->
+                Mono.error(
+                    InvalidSearchCriteriaException(
+                        TransactionDataProvider.SearchTypeMapping.getSearchType(
+                            searchCriteria.javaClass
+                        ),
```

Comment:
I do feel this is more complicated than it should be. Can't we use:
```suggestion
                        searchCriteria.type,
```
?

Reply:
Yep, I overdone here :D fix with https://github.com/pagopa/pagopa-ecommerce-helpdesk-service/pull/9/commits/1bba1565062d346fa52598efca17614066eecc77